u_parrots.go

```diff
@@ -17,26 +17,18 @@ import (
 )
 
 var ErrUnknownClientHelloID = errors.New("tls: unknown ClientHelloID")
-var ErrNotPSKClientHelloID = errors.New("tls: ClientHello does not contain pre_shared_key extension")
-var ErrPSKExtensionExpected = errors.New("tls: pre_shared_key extension expected when fetching preset ClientHelloSpec")
 
 // UTLSIdToSpec converts a ClientHelloID to a corresponding ClientHelloSpec.
 //
 // Exported internal function utlsIdToSpec per request.
-func UTLSIdToSpec(id ClientHelloID, pskExtension ...PreSharedKeyExtension) (ClientHelloSpec, error) {
-	if len(pskExtension) > 1 {
-		return ClientHelloSpec{}, errors.New("tls: at most one PreSharedKeyExtensions is allowed")
-	}
-
-	chs, err := utlsIdToSpec(id)
-	if err != nil && errors.Is(err, ErrUnknownClientHelloID) {
-		chs, err = utlsIdToSpecWithPSK(id, pskExtension...)
-	}
-
-	return chs, err
+func UTLSIdToSpec(id ClientHelloID, pskExt PreSharedKeyExtension, sessionTicketExt *SessionTicketExtension) (ClientHelloSpec, error) {
+	return utlsIdToSpec(id, pskExt, sessionTicketExt)
 }
 
-func utlsIdToSpec(id ClientHelloID) (ClientHelloSpec, error) {
+func utlsIdToSpec(id ClientHelloID, pskExt PreSharedKeyExtension, sessionTicketExt *SessionTicketExtension) (ClientHelloSpec, error) {
+	if pskExt == nil || sessionTicketExt == nil {
```

Comment:
This might not always hold true, since it is totally valid to have a `ClientHelloSpec` without using any of `PreSharedKeyExtension` or `SessionTicketExtension`.

Reply:
reverted

Reply:
I saw you reverted it back to `utlsIdToSpec(id ClientHelloID) (ClientHelloSpec, error)`, but it might be useful to also have ways to get a `ClientHelloSpec` with specified `*SessionTicketExtension` and/or `PreSharedKeyExtension`. 

I don't have a good clean way to redesign the interface yet, so possibly 

```
func utlsIdToSpec(id ClientHelloID) (ClientHelloSpec, error) {
    return utlsIdToSpecAdvanced(id, nil, nil)
}

func utlsIdToSpecAdvanced(ClientHelloID, *SessionTicketExtension, PreSharedKeyExtension) {
    // ...
}
```

Assuming similar thing for `ApplyPreset()` and `UTLSIdToSpec()` should be needed. 

Or maybe you already have something similar to this that I missed?

Reply:
Nvm. Saw your last comment.